src/midgard/polyline2.cc

```diff
@@ -91,14 +91,18 @@ template <class container_t>
 void Polyline2<coord_t>::Generalize(container_t& polyline,
                                     float epsilon,
                                     const std::unordered_set<size_t>& indices) {
+  // any epsilon this low will have no effect on the input nor will any super short input
+  if (epsilon <= 0.f || polyline.size() < 3)
+    return;
+
   // the recursive bit
   epsilon *= epsilon;
   std::function<void(typename container_t::iterator, size_t, typename container_t::iterator, size_t)>
       peucker;
   peucker = [&peucker, &polyline, epsilon, &indices](typename container_t::iterator start, size_t s,
                                                      typename container_t::iterator end, size_t e) {
     // find the point furthest from the line
-    float dmax = 0.f;
+    float dmax = std::numeric_limits<float>::lowest();
```

Comment:
if you did want to remove the `if` above you need to use a number lower than any other number because otherwise the `if` below will think you found points to simplify when you didn't